### PR TITLE
Work on organization structure

### PR DESCRIPTION
**Pull request by 2HEADSs**
Add part of Organization service, controller and model. Still have a lot of questions to finish the task? There is a lot of TODO's wich will be fixed after comunication with the team!

### CODE REVIEW

File: src/models/User.js [hunk 1]
```diff
@@ -1,32 +1,80 @@
...
-    email: {
-        type: String, required: true, unique: true, validate: {
-            validator: (value) => emailRegex.test(value),
-            message: "Invalid email",
-
-        }
+  email: {
+    type: String,
+    required: true,
+    unique: true,
```
Comment (Getsov) [inline src/models/User.js:16]: Is this working?

File: src/models/User.js [hunk 2]
```diff
@@ -1,32 +1,80 @@
...
+  //TODO:  minimum and maximum length lastName
+  lastName: {
+    type: String,
+    require: true,
+    minlength: [2, "Last name must be minimum 2 characters long!"],
+    maxlength: [15, "Last name must be maximum 15 characters long!"],
+  },
+  role: {
+    type: String,
+    require: true,
```
Comment (Getsov) [inline src/models/User.js:38]: Shouldn’t be required?

File: src/models/User.js [hunk 3]
```diff
@@ -1,32 +1,80 @@
...
+  firstName: {
+    type: String,
+    require: true,
+    minlength: [2, "First name must be minimum 2 characters long!"],
+    maxlength: [15, "First name must be maximum 15 characters long!"],
+  },
+  //TODO:  minimum and maximum length lastName
+  lastName: {
+    type: String,
+    require: true,
```
Comment (Getsov) [inline src/models/User.js:32]: Shouldn’t be required?

File: src/models/User.js [hunk 4]
```diff
@@ -1,32 +1,80 @@
...
+      message: "Invalid email",
     },
-    firstName: { type: String, require: true },
-    lastName: { type: String, require: true },
-    isDeleted: {type: Boolean, default: false}
+  },
+  //TODO:  minimum and maximum length firstName
+  firstName: {
+    type: String,
+    require: true,
```
Comment (Getsov) [inline src/models/User.js:25]: Shouldn’t be required?

**Review by Getsov — COMMENTED** [review]
A lot of comments in the code, it’s hard to read and follow. Don’t add more features until the comments are cleared

File: src/models/User.js (hunk 3 above)
Comment (2HEADSs) [reply src/models/User.js:32]: Yes, my mistake! I will fix it in next commit!

File: src/models/User.js (hunk 2 above)
Comment (2HEADSs) [reply src/models/User.js:38]: Yes, my mistake! I will fix it in next commit!

File: src/models/User.js (hunk 4 above)
Comment (2HEADSs) [reply src/models/User.js:25]: Yes, my mistake! I will fix it in next commit!

File: src/models/User.js (hunk 1 above)
Comment (2HEADSs) [reply src/models/User.js:16]: I'm not sure what you talking about but return the propper response.

Comment (Getsov) [reply src/models/User.js:16]: I mean if you try to add two identical emails, is it working?

**Review by Getsov — APPROVED** [review]

**Pull request merged**